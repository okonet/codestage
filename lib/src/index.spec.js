const { highlight, renderRTF } = require('./index')

describe('renderRTF', () => {
  it('should not add unnecessary new lines', async () => {
    const html = ''
    const css = ''
    const res = renderRTF(html, css)
    expect(res).toMatchSnapshot()
  })

  it('should apply simple CSS selectors to HTML', async () => {
    const html = '<span class="a">test</span>'
    const css = `
     .a { color: red; }
    `
    const res = renderRTF(html, css)
    expect(res).toMatchSnapshot()
  })

  it('should apply complex CSS selectors to HTML', async () => {
    const html = '<span class="a b">test</span>'
    const css = `
     .a { color: red; }
     .a.b { color: blue; }
    `
    const res = renderRTF(html, css)
    expect(res).toMatchSnapshot()
  })

  it('should apply even more complex CSS selectors to HTML', async () => {
    const html = `
      <div class="wrapper">
        <span class="c">1</span>
        <span class="a">2</span>
        <span class="a b">3</span>
      </div>`
    const css = `
     .wrapper { color: rgb(0,255,0); }
     .a { color: red; }
     .a.b { color: blue; }
    `
    const res = renderRTF(html, css)
    expect(res).toMatchSnapshot()
  })
})

describe('highlight renderer', () => {
  it('should not add unnecessary new lines', async () => {
    const source = ''
    const res = await highlight(source, 'javascript')
    expect(res.value).toMatchSnapshot()
  })

  it('should render multiple lines', async () => {
    const source = `line1
line2


line5
`
    const res = await highlight(source, 'javascript')
    expect(res.value).toMatchSnapshot()
  })

  it('should render a string', async () => {
    const source = 'const a = "foo";'
    const res = await highlight(source, 'javascript', { theme: 'chrome' })
    expect(res.value).toMatchSnapshot()
  })

  it('should render multiple lines', async () => {
    const source = `line1
line2


line5
`
    const res = await highlight(source, 'javascript')
    expect(res.value).toMatchSnapshot()
  })

  it('should encode curly brackets and back slashes', async () => {
    const source = `{
    \\ test
    }`
    const res = await highlight(source, 'javascript')
    expect(res.value).toMatchSnapshot()
  })

  it('should not add white spaces', async () => {
    const source = `test.then((result) => {
  console.log(result.stdout)
})`
    const res = await highlight(source, 'javascript')
    expect(res.value).toMatchSnapshot()
  })

  it('should render specific styles', async () => {
    const source = `
// Comment
function test() {}
`
    const res = await highlight(source, 'javascript', { theme: 'xcode' })
    expect(res.value).toMatchSnapshot()
  })

  it('should render javascript', async () => {
    const source = `settings.set('user.name', {
  first: 'Cosmo',
  last: 'Kramer'
}).then(() => {
  settings.get('user.name.first').then(val => {
    console.log(val);
    // => 'Cosmo'
  });
});
`
    const res = await highlight(source, 'javascript', { theme: 'xcode' })
    expect(res.value).toMatchSnapshot()
  })

  it('should encode single unicode symbols', async () => {
    // RTF doesn't suport Unicode so we need to encode them
    const source = 'â € ¤ каждой'
    const res = await highlight(source, 'css')
    expect(res.value).toMatchSnapshot()
  })

  it('should encode double unicode symbols', async () => {
    // RTF doesn't suport Unicode so we need to encode them
    const source = 'က 힣 ↕ ↕︎ 鼖'
    const res = await highlight(source, 'css')
    expect(res.value).toMatchSnapshot()
  })

  it('should not throw on unparsed colors', async () => {
    expect.assertions(1)
    const source = '* { background: url("image.png"); }'
    await expect(highlight(source, 'css')).resolves.toBeDefined()
  })
})
