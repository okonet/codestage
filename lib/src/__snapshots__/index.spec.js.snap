// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlightRTFRenderer should encode curly brackets and back slashes 1`] = `
"{\\\\rtf1\\\\ansi\\\\deff0
{\\\\fonttbl{\\\\f0\\\\fmodern\\\\fprq1\\\\fcharset0 Courier New;}}
{\\\\colortbl;\\\\red255\\\\green255\\\\blue255;\\\\red0\\\\green106\\\\blue0;\\\\red170\\\\green13\\\\blue145;\\\\red0\\\\green0\\\\blue136;\\\\red102\\\\green102\\\\blue0;\\\\red196\\\\green26\\\\blue22;\\\\red0\\\\green136\\\\blue0;\\\\red28\\\\green0\\\\blue207;\\\\red92\\\\green38\\\\blue153;\\\\red0\\\\green0\\\\blue0;\\\\red238\\\\green238\\\\blue238;\\\\red186\\\\green238\\\\blue186;\\\\red255\\\\green200\\\\blue189;\\\\red155\\\\green112\\\\blue63;}
\\\\f0\\\\fs40
\\\\{\\\\par
    \\\\\\\\ test\\\\par
    \\\\}
}"
`;

exports[`highlightRTFRenderer should not add white spaces 1`] = `
"{\\\\rtf1\\\\ansi\\\\deff0
{\\\\fonttbl{\\\\f0\\\\fmodern\\\\fprq1\\\\fcharset0 Courier New;}}
{\\\\colortbl;\\\\red255\\\\green255\\\\blue255;\\\\red0\\\\green106\\\\blue0;\\\\red170\\\\green13\\\\blue145;\\\\red0\\\\green0\\\\blue136;\\\\red102\\\\green102\\\\blue0;\\\\red196\\\\green26\\\\blue22;\\\\red0\\\\green136\\\\blue0;\\\\red28\\\\green0\\\\blue207;\\\\red92\\\\green38\\\\blue153;\\\\red0\\\\green0\\\\blue0;\\\\red238\\\\green238\\\\blue238;\\\\red186\\\\green238\\\\blue186;\\\\red255\\\\green200\\\\blue189;\\\\red155\\\\green112\\\\blue63;}
\\\\f0\\\\fs40
test.then(({\\\\cf8 result}) => \\\\{\\\\par
  {\\\\cf8 console}.log(result.stdout)\\\\par
\\\\})
}"
`;

exports[`highlightRTFRenderer should render a string 1`] = `
"{\\\\rtf1\\\\ansi\\\\deff0
{\\\\fonttbl{\\\\f0\\\\fmodern\\\\fprq1\\\\fcharset0 Courier New;}}
{\\\\colortbl;\\\\red255\\\\green255\\\\blue255;\\\\red0\\\\green106\\\\blue0;\\\\red170\\\\green13\\\\blue145;\\\\red0\\\\green0\\\\blue136;\\\\red102\\\\green102\\\\blue0;\\\\red196\\\\green26\\\\blue22;\\\\red0\\\\green136\\\\blue0;\\\\red28\\\\green0\\\\blue207;\\\\red92\\\\green38\\\\blue153;\\\\red0\\\\green0\\\\blue0;\\\\red238\\\\green238\\\\blue238;\\\\red186\\\\green238\\\\blue186;\\\\red255\\\\green200\\\\blue189;\\\\red155\\\\green112\\\\blue63;}
\\\\f0\\\\fs40
{\\\\cf2 const} a = {\\\\cf5 \\"foo\\"};
}"
`;

exports[`highlightRTFRenderer should render javascript 1`] = `
"{\\\\rtf1\\\\ansi\\\\deff0
{\\\\fonttbl{\\\\f0\\\\fmodern\\\\fprq1\\\\fcharset0 Courier New;}}
{\\\\colortbl;\\\\red51\\\\green51\\\\blue51;\\\\red248\\\\green248\\\\blue248;\\\\red153\\\\green153\\\\blue136;\\\\red0\\\\green128\\\\blue128;\\\\red221\\\\green17\\\\blue68;\\\\red153\\\\green0\\\\blue0;\\\\red68\\\\green85\\\\blue136;\\\\red0\\\\green0\\\\blue128;\\\\red0\\\\green153\\\\blue38;\\\\red153\\\\green0\\\\blue115;\\\\red0\\\\green134\\\\blue179;\\\\red153\\\\green153\\\\blue153;\\\\red255\\\\green221\\\\blue221;\\\\red221\\\\green255\\\\blue221;}
\\\\f0\\\\fs40
settings.set({\\\\cf4 'user.name'}, \\\\{\\\\par
  first: {\\\\cf4 'Cosmo'},\\\\par
  last: {\\\\cf4 'Kramer'}\\\\par
\\\\}).{\\\\cf0\\\\b then}(() => \\\\{\\\\par
  settings.get({\\\\cf4 'user.name.first'}).{\\\\cf0\\\\b then}(val => \\\\{\\\\par
    {\\\\cf10 console}.log(val);\\\\par
    {\\\\cf8 //} => {\\\\cf4 'Cosmo'}\\\\par
  \\\\});\\\\par
\\\\});\\\\par

}"
`;

exports[`highlightRTFRenderer should render multiple lines 1`] = `
"{\\\\rtf1\\\\ansi\\\\deff0
{\\\\fonttbl{\\\\f0\\\\fmodern\\\\fprq1\\\\fcharset0 Courier New;}}
{\\\\colortbl;\\\\red255\\\\green255\\\\blue255;\\\\red0\\\\green106\\\\blue0;\\\\red170\\\\green13\\\\blue145;\\\\red0\\\\green0\\\\blue136;\\\\red102\\\\green102\\\\blue0;\\\\red196\\\\green26\\\\blue22;\\\\red0\\\\green136\\\\blue0;\\\\red28\\\\green0\\\\blue207;\\\\red92\\\\green38\\\\blue153;\\\\red0\\\\green0\\\\blue0;\\\\red238\\\\green238\\\\blue238;\\\\red186\\\\green238\\\\blue186;\\\\red255\\\\green200\\\\blue189;\\\\red155\\\\green112\\\\blue63;}
\\\\f0\\\\fs40
line1\\\\par
line2\\\\par
\\\\par
\\\\par
line5\\\\par

}"
`;

exports[`highlightRTFRenderer should render specific styles 1`] = `
"{\\\\rtf1\\\\ansi\\\\deff0
{\\\\fonttbl{\\\\f0\\\\fmodern\\\\fprq1\\\\fcharset0 Courier New;}}
{\\\\colortbl;\\\\red0\\\\green0\\\\blue0;\\\\red255\\\\green255\\\\blue255;\\\\red128\\\\green128\\\\blue128;\\\\red128\\\\green128\\\\blue0;\\\\red239\\\\green239\\\\blue239;\\\\red0\\\\green0\\\\blue128;\\\\red0\\\\green0\\\\blue255;\\\\red0\\\\green128\\\\blue0;\\\\red208\\\\green237\\\\blue237;\\\\red102\\\\green14\\\\blue122;\\\\red186\\\\green238\\\\blue186;\\\\red255\\\\green200\\\\blue189;}
\\\\f0\\\\fs40
\\\\par
{\\\\cf2\\\\i // Comment}\\\\par
{\\\\b\\\\cf5 function} {\\\\cf0 test}() \\\\{\\\\}\\\\par

}"
`;
